roxctl/scanner/generate/generate.go

```diff
@@ -16,46 +17,81 @@ import (
 	"github.com/stackrox/rox/roxctl/scanner/clustertype"
 )
 
-// Options stores options related to scanner generate commands.
-type Options struct {
-	OutputDir string
+const (
+	scannerGenerateAPIPath = "/api/extensions/scanner/zip"
+)
+
+type scannerGenerateCommand struct {
+	// Properties that are bound to cobra flags.
+	outputDir string
+	apiParams apiparams.Scanner
+	timeout   time.Duration
+}
+
+func (cmd *scannerGenerateCommand) construct(c *cobra.Command) {
+	cmd.timeout = flags.Timeout(c)
+}
+
+func (cmd *scannerGenerateCommand) validate() error {
+	// Validate supported Istio versions.
+	if cmd.apiParams.IstioVersion != "" {
+		for _, istioVersion := range istioutils.ListKnownIstioVersions() {
+			if cmd.apiParams.IstioVersion == istioVersion {
+				return nil
+			}
+		}
+
+		return errors.New("unsupported Istio version")
```

Comment:
maybe it's good to say what version?
How about using errox error with invalid arg?

Reply:
✅